CommonCompilerConfig.cmake

```diff
@@ -102,6 +104,22 @@ endif()
 message(STATUS "CMAKE BITNESS=${BITNESS}")
 
 
+# Check memory limits (in megabytes)
+if(CMAKE_MAJOR_VERSION GREATER 2)
+  cmake_host_system_information(RESULT total_physical_memory QUERY TOTAL_PHYSICAL_MEMORY)
+  cmake_host_system_information(RESULT total_virtual_memory QUERY TOTAL_VIRTUAL_MEMORY)
+  math(EXPR total_memory "${total_physical_memory}+${total_virtual_memory}")
```

Comment:
As you mentioned above, if you don't have any swap file then compilation can be impossible with insufficient physical memory. Adding them together at least gives an indication of whether a lack of physical memory can be assisted with an available swap file.